pulsar-client/src/main/java/org/apache/pulsar/client/impl/conf/ConsumerConfigurationData.java

```diff
@@ -149,12 +147,13 @@ public int getMaxPendingChuckedMessage() {
 
     private boolean resetIncludeHead = false;
 
+    @JsonIgnore
```

Comment:
Can you add some details about why we add `@JsonIgnore ` here?

Reply:
KeySharedPolicy is not deserializable:
```java
    @Test
    public void test() {
        Map<String, Object> config = new HashMap<>();
        config.put("keySharedPolicy", KeySharedPolicy.autoSplitHashRange());
        consumerBuilderImpl.loadConf(config);
    }
```
fails with `com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of org.apache.pulsar.client.api.KeySharedPolicy`